SNOW-227101: fix to datetime columns being created without timezone

Nazarii opened the pull request:
[Original ticket.](https://github.com/snowflakedb/snowflake-sqlalchemy/issues/201)

Commit 28e8031 introduced a bug where all `sqlalchemy.DateTime(timezone=True)` columns were created as `DATETIME` in Snowflake which is an alias to `TIMESTAMP_NTZ`.
This commit reverts datetime logic back so if the timezone is defined in datetime column it will create `TIMESTAMP_TZ` and `TIMESTAMP_NTZ` otherwise.
This bug was introduced in [v1.1.18](https://github.com/snowflakedb/snowflake-sqlalchemy/commit/28e80317968b51da401803fb98515e5e70a35885)

WeiGuHS commented:
@Nazarii do you know when this commit can be merged? and also is a reason that take out these TIMESTAMP_TZ from v1.1.8?

Nazarii commented:
@WeiGuHS no idea, this PR has no reviewers assigned, the same with the original ticket. I have no permissions to assign PR, I'll try to ping other contributors to review. It looks like the support of `TIMESTAMP_TZ` was removed by a mistake.

Nazarii commented:
I have read the CLA Document and I hereby sign the CLA

Nazarii commented:
Not sure why but this PR shows there are conflicts. Diff does not show any merge conflicts or maybe there are some other conflicts?

sfc-gh-mkeller commented:
We will revisit this issue shortly.
In the meantime `TIMESTAMP` is an alias that is configurable actually, please see [our docs](https://docs.snowflake.com/en/sql-reference/parameters.html#timestamp-type-mapping) to see how to change what it points at.

philippeboyd commented:
@sfc-gh-mkeller any news on this?

conflicts comes from GEOGRAPHY being added at the same spot in the file as your modifications

```python
    def visit_GEOGRAPHY(self, type_, **kw):
        return "GEOGRAPHY"
```

0lmer committed:
SNOW-227101: fix to datetime columns being created without timezone (…
…TIMESTAMP_NTZ).

Commit 28e8031 introduced a bug where all sqlalchemy.DateTime(timezone=True) columns were created as DATETIME in Snowflake which is an alias to TIMESTAMP_NTZ.
This commit reverts datetime logic back so if the timezone is defined in datetime column it will create TIMESTAMP_TZ and TIMESTAMP_NTZ otherwise.

(cherry picked from commit e180b23)

sfc-gh-mraba commented:
Closing — superseded by #649 ("Add support for timestamps with timezone"). On `main`, timezone-aware `DateTime`/`DATETIME` render as Snowflake's `TIMESTAMP_TZ` (and `TIMESTAMP_NTZ`/`LTZ` are supported), rather than the `TIMESTAMP WITH TIME ZONE` form proposed here. Thanks @Nazarii.
